### PR TITLE
Better implementation of displaying textDocument/signatureHelp

### PR DESCRIPTION
**Pull request by yyoncho**
This implementation is more accurate and closer to the lsp protocol idea.

- the signatures could be browsed using `M-n`/`M-p` and `C-c C-k` to switch back
to hover info

- the signatureHelp is triggered when

- added two new variables: `lsp-signature-render-documentation` - to control
whether to include signature documentation and `lsp-signature-auto-activate`.
When `lsp-signature-auto-activate` is true the signatureHelp will be displayed
when you press the signature trigger char(e. g. `(` in java).

``` elisp
(add-hook 'company-completion-finished-hook (lambda (&rest _) (lsp-signature-activate)))
```

### CODE REVIEW

File: README.org [hunk 1]
```diff
@@ -202,6 +202,10 @@
    - ~lsp-lens-hide~ - Hide lenses in the current file
    - ~lsp-lens-mode~  - Turn on/off lenses in the current file.
    - ~lsp-avy-lens~ - Click lens using ~avy~
+   - ~lsp-signature-activate~ - activate signature help. Bound to ~C-M-SPC~ by default.
```
Comment (seagle0128) [inline README.org]: The keybinding is `C-S-SPC` in `lsp-mode-map`.

File: lsp-mode.el [hunk 2]
```diff
@@ -125,18 +125,6 @@
   :type 'boolean
   :package-version '(lsp-mode . "6.1"))
 
-(defcustom lsp-use-native-json t
```
Comment (ocharles) [inline lsp-mode.el]: This seems to be the PR that removes this option. Do those of us on Emacs HEAD need to do anything to get native JSON parsing?

Comment (yyoncho) [reply lsp-mode.el]: No, it will be picked automatically. The flag was there temporary as a fallback option when we introduced native json support.

Comment (ocharles) [reply lsp-mode.el]: Thank you for the very fast response! I look forward to trying `lsp-mode` in Emacs HEAD then! :smile:

Comment (yyoncho) [reply lsp-mode.el]: Make sure you remove the `elc` generated from the old emacs - we do some evaluations compile time depending on emacs version to make the runtime faster. (The easiest option is to delete the package and install it again).

Comment (honnix) [reply lsp-mode.el]: I wish I could see this comment earlier. Thanks!

**Pull request merged**